neurogym/envs/contextdecisionmaking.py

```diff
@@ -31,27 +39,36 @@ class SingleContextDecisionMaking(ngym.TrialEnv):
     def __init__(
         self,
         dt=100,
-        context=0,
+        explicit_context=False,
+        fixed_context=0,
+        dim_ring=2,
         rewards=None,
         timing=None,
         sigma=1.0,
-        dim_ring=2,
     ) -> None:
         super().__init__(dt=dt)
 
-        # trial conditions
+        # Task parameters
+        self.explicit_context = explicit_context
+        self.context = fixed_context if not explicit_context else None  # Store fixed context for implicit case
+        self.dim_ring = dim_ring
+        self.sigma = sigma / np.sqrt(self.dt)
+
+        # Trial conditions
         self.cohs = [5, 15, 50]
-        self.sigma = sigma / np.sqrt(self.dt)  # Input noise
-        self.context = context
+        if explicit_context:
+            self.contexts = [0, 1]
+        else:
+            self.contexts = [self.context]
```

Comment:
this line is flagged by mypy locally (self.contexts not the expected type)
I noticed that the static typing action did not run on this PR. Any idea why?

Reply:
My local mypy run now shows no errors. I fixed the issue in the YAML files (we were previously running the action only when the PR was targeting main). Now, it runs every time.